lib/engine-addon.js

```diff
@@ -281,6 +282,15 @@ var buildEngineStyleTree = memoize(function buildEngineStyleTree() {
   return this.debugTree(combinedEngineStylesAndDependencyStylesTree, 'engine-style:input');
 });
 
+/**
+ * Generates an identifier for an addon, which will be used when
+ * addons are deduped. Different instances of the same addon will
+ * be detected based on this identifier.
+ */
+function generateAddonId(addon) {
```

Comment:
The initial idea was to encapsulate a better algorithm to determine the duplicate addon, which would be based on something more than just the name. However, considering that ember-cli doesn't have any notion of addon versioning in the same project, I think it was unnecessary. I will switch back to just use the name of the addon.